facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -192,23 +224,81 @@ private void processDeliverables(
             build.getArtifacts()
                     .stream()
                     .peek(statCounter)
-                    .map(artifactParser)
-                    .peek(artifactUpdater)
-                    .forEach(milestone::addDeliveredArtifact);
+                    .forEach(
+                            artifactDto -> {
+                                addDeliveredArtifact(
+                                    effectivlyFinalArtifactParser.apply(artifactDto),
+                                    report,
+                                    artifactDto.isBuiltFromSource(),
+                                    build.getBrewId());
+                            });
         }
+
         if (!notFoundArtifacts.isEmpty()) {
             TargetRepository distributionRepository = getDistributionRepository(distributionUrl);
             notFoundArtifacts.stream()
                     .peek(stats.notFoundCounter())
                     .map(art -> findOrCreateNotFoundArtifact(art, distributionRepository))
-                    .peek(artifactUpdater)
-                    .forEach(milestone::addDeliveredArtifact);
+                    .forEach(artifact -> addDeliveredArtifact(artifact, report, false, null));
         }
         stats.log(distributionUrl);
-        milestone.setDeliveredArtifactsImporter(userService.currentUser());
+    }
+
+    private void addDeliveredArtifact(
+            org.jboss.pnc.model.Artifact artifact,
+            DeliverableAnalyzerReport report,
+            boolean builtFromSource,
+            Long brewBuildId) {
+        DeliverableArtifact deliverableArtifact = DeliverableArtifact.builder()
+                .artifact(artifact)
+                .report(report)
+                .builtFromSource(builtFromSource)
+                .brewBuildId(brewBuildId)
+                .build();
+        report.addDeliverableArtifact(deliverableArtifact);
+        deliverableArtifactRepository.save(deliverableArtifact);
+    }
+
+    private DeliverableAnalyzerReport createReportForCompletedAnalysis(
+            Base32LongID operationId,
+            boolean wasRunAsScratchAnalysis) {
+        org.jboss.pnc.model.DeliverableAnalyzerOperation operation = deliverableAnalyzerOperationRepository
+                .queryById(operationId);
+        operation.setUser(userService.currentUser());
+
+        var report = DeliverableAnalyzerReport.builder()
+                .id(operationId)
+                .operation(operation)
+                .labels(getReportLabels(wasRunAsScratchAnalysis))
+                .labelHistory(new ArrayList<>())
+                .artifacts(new HashSet<>(Set.of()))
+                .build();
+        deliverableAnalyzerReportRepository.save(report);
+        if (wasRunAsScratchAnalysis) {
+            updateLabelHistoryWithScratchEntry(report);
+        }
+        return report;
+    }
+
+    private EnumSet<DeliverableAnalyzerReportLabel> getReportLabels(boolean wasRunAsScratchAnalysis) {
```

Comment:
I like that you extracted this to method :+1:

Reply:
:heart: